include/openthread/tcp.h

```diff
@@ -0,0 +1,616 @@
+otError otTcpBind(otInstance *aInstance, otTcpEndpoint *aEndpoint, const otSockAddr *aSockName);
+
+/**
+ * This enumeration defines flags passed to @p otTcpConnect.
+ */
+enum {
+    OT_TCP_CONNECT_NO_FAST_OPEN = 1 << 0,
+};
+
+/**
+ * Records the remote host and port for this connection. By default TCP Fast
+ * Open is used. This means that this function merely records the remote host
+ * and port, and that the TCP connection establishment handshake only happens
+ * on the first call to @p otTcpSendByReference. TCP Fast Open can be
+ * explicitly disabled using @p aFlags, in which case the TCP connection
+ * establishment handshake is initiated immediately.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint  A pointer to the TCP endpoint structure to connect.
+ * @param[in]  aSockName  The IP address and port of the host to which to connect.
+ * @param[in]  aFlags     Flags specifying options for this operation (see enumeration above).
+ *
+ * @retval OT_ERROR_NONE    Successfully completed the operation.
+ * @retval OT_ERROR_FAILED  Failed to complete the operation.
+ */
+otError otTcpConnect(otInstance *aInstance, otTcpEndpoint *aEndpoint, const otSockAddr *aSockName, uint32_t aFlags);
+
+/**
+ * This enumeration defines flags passed to @p otTcpSendByReference.
+ */
+enum {
+    OT_TCP_SEND_MORE_TO_COME = 1 << 0,
+};
+
+/**
+ * Adds data referenced by the linked buffer pointed to by @p aBuffer to the
+ * the send buffer. Upon a sucessful call to this function, the linked buffer
+ * and data it references are owned by the TCP stack; they should not be
+ * modified by the application until a "send done" callback returns ownership
+ * of those objects to the application. It is acceptable to call this function
+ * to add another linked buffer to the send queue, even if the "send done"
+ * callback for a previous invocation of this function has not yet fired.
+ *
+ * Note that @p aBuffer should not be chained; its @p mNext field should be
+ * @p NULL. If additional data will be added right after this call, then the
+ * @p OT_TCP_SEND_MORE_TO_COME flag should be used as a hint to the TCP
+ * implementation.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint  A pointer to the TCP endpoint structure representing the TCP endpoint on which to send data.
+ * @param[in]  aBuffer    A pointer to the linked buffer chain referencing data to add to the send buffer.
+ * @param[in]  aFlags     Flags specifying options for this operation (see enumeration above).
+ *
+ * @retval OT_ERROR_NONE    Successfully added data to the send buffer.
+ * @retval OT_ERROR_FAILED  Failed to add data to the send buffer.
+ */
+otError otTcpSendByReference(otInstance *aInstance, otTcpEndpoint *aEndpoint, otLinkedBuffer *aBuffer, uint32_t aFlags);
+
+/**
+ * Adds data to the send buffer by extending the length of the final
```

Comment:
nit: have brief in separate block above

Reply:
Done. I'm not 100% what you meant by "in own block", can you please check that what I've done is acceptable?